File: client-scalaz/src/main/scala/org/openapitools/client/api/FilterResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import FilterResponse._

case class FilterResponse (
  filterId: Option[Long],
name: Option[String],
display: Option[String],
subFilterCount: Option[Integer],
parentId: Option[Long],
description: Option[String],
active: Option[Boolean],
externalId: Option[String],
externalType: Option[String])

object FilterResponse {
  import DateTimeCodecs._

  implicit val FilterResponseCodecJson: CodecJson[FilterResponse] = CodecJson.derive[FilterResponse]
  implicit val FilterResponseDecoder: EntityDecoder[FilterResponse] = jsonOf[FilterResponse]
  implicit val FilterResponseEncoder: EntityEncoder[FilterResponse] = jsonEncoderOf[FilterResponse]
}

File: client-clojure/src/sirqul_io_t_platform/specs/filter_response.clj
(ns sirqul-io-t-platform.specs.filter-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            )
  (:import (java.io File)))


(def filter-response-data
  {
   (ds/opt :filterId) int?
   (ds/opt :name) string?
   (ds/opt :display) string?
   (ds/opt :subFilterCount) int?
   (ds/opt :parentId) int?
   (ds/opt :description) string?
   (ds/opt :active) boolean?
   (ds/opt :externalId) string?
   (ds/opt :externalType) string?
   })

(def filter-response-spec
  (ds/spec
    {:name ::filter-response
     :spec filter-response-data}))
